evaluator/evaluator/routers/dataset.py

```diff
@@ -1,22 +1,234 @@
+                "value": {
+                    "name": "commensense_qa",
+                    "skill_type": "multiple-choice",
+                    "metric": "accuracy",
+                    "mapping": {
+                        "id-column": "id",
+                        "question-column": "question",
+                        "choice-columns": ["choice.text"],
+                        "choices-key-mapping-column": "choices.label",
+                        "answer-index-column": "answerkey",
+                    },
+                },
+            },
+            "Extractivesample": {
+                "summary": "Extractivesample",
+                "description": "for Extractive Dataset",
+                "value": {
+                    "name": "quoref",
+                    "skill-type": "extractive-qa",
+                    "metric": "squad",
+                    "mapping": {
+                        "id-column": "id",
+                        "question-column": "question",
+                        "context-column": "context",
+                        "answer-text-column": "answers.text",
+                    },
+                },
+            },
+        },
+    ),
+):
+    logger.debug(f"post dataset: dataset_name= {name}")
+    dataset_name_exist = mongo_client.client.evaluator.datasets.find(
+        {"dataset_name": name}
+    )
+    logger.debug(f"post anfrage")
+    if dataset_name_exist is None:
+        logger.debug(f"dataset name is None ")
+
+        if name == "quoref":
+            dataset_mapping = {
+                "dataset_name": name,
+                "skill-type": skill_type,
+                "metric": metric,
+                "mapping": {
+                    "id-column": mapping["id-column"],
+                    "question-column": mapping["question-column"],
+                    "context-column": mapping["context-column"],
+                    "answer-text-column": mapping["answer-text-column"],
+                },
+            }
+
+        elif name == "commensense_qa":
+            dataset_mapping = {
+                "dataset_name": name,
+                "skill-type": skill_type,
+                "metric": metric,
+                "mapping": {
+                    "id-column": mapping["id-column"],
+                    "question-column": mapping["question-column"],
+                    "choice-columns": mapping["choice-columns"],
+                    "choices-key-mapping-column": mapping["choices-key-mapping-column"],
+                    "answer-index-column": mapping["answer-index-column"],
+                },
+            }
+
+            dataset_ob = Dataset(
+                dataset_name=dataset_name,
+                skill_type=skill_type,
+                metric=metric,
+                mapping=mapping,
+            )
+            # mongo_client.client.evaluator.datasets.insert_one(dataset_mapping)
+            mongo_client.client.evaluator.datasets.insert_one(dataset_mapping)
```

Comment:
Hier musst du doch jetzt dein `Dataset` inserten 🤔 Also dataset_obj (bzw. da das ja redundant ist einfach das Dataset aus dem request)

Reply:
ich habe die Anpassung gemacht

Reply:
Die insert-Methode funktioniert nicht  mit dataset object. Es muss ein dictionary sein. Dafür habe ich dataset_mapping definiert. Also bei mir hatte ich eine Fehlermeldung bekommen als ich das Dataset-Objekt benutzt hatte

Reply:
Du musst `Dataset` (in models.py) einfach zu einen `MongoModel` statt einem `BaseModel` machen (vielleicht musst du das auch für `MultipleChoiceDatasetSample` und `ExtractiveDatasetSample` anpassen). Und dann kannst du einfach folgendes machen... : `mongo_client.client.evaluator.datasets.insert_one(dataset.mongo())`. 

`.mongo()` wandelt das dann in ein dictionary um für dich.
Und beim Auslesen kannst du `Dataset.from_mongo(...)` benutzen.

Reply:
ja, ich habe die Anpassung gemacht und es funktioniert. Danke